docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/TransferLearning.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using ICSharpCode.SharpZipLib.GZip;
+using ICSharpCode.SharpZipLib.Tar;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    public static class TransferLearning
+    {
+        /// <summary>
+        /// Example use of the TensorFlow image model in a ML.NET pipeline.
+        /// </summary>
+        public static void Example()
+        {
+            var sw = new Stopwatch();
+
+            var mlContext = new MLContext();
+            var data = GetTensorData();
+            var idv = mlContext.Data.LoadFromEnumerable(data);
+
+            // Create a ML pipeline.
+            var pipeline = 
+                mlContext.Transforms.Conversion.MapValueToKey(
+                    nameof(TensorData.Label))
+                .Append(mlContext.Model.ImageClassification(
+                    nameof(TensorData.input),
+                    nameof(TensorData.Label), batchSize: 2,
+                        addBatchDimensionInput: true));
+
+            // Run the pipeline and get the transformed values.
+            var estimator = pipeline.Fit(idv);
+            var transformedValues = estimator.Transform(idv);
```

Comment:
For this example and making batch predictions with Transform() I think we should provide a different TEST IDataView instead of using the same IDataView used for training too?

Reply:
Ideally but this is just a sample and we have done similar things in other samples too in the project. Its point is to illustrate the idea in a very simple manner.

---
In reply to: [310269964](https://github.com/dotnet/machinelearning/pull/4057#discussion_r310269964) [](ancestors = 310269964)